FrontEnd/src/ProfilePage/UserProfile.jsx

```diff
@@ -1,19 +1,51 @@
-import { Link } from "react-router-dom";
+import { useNavigate } from "react-router-dom";
 import { useEffect, useState } from "react";
 import { logOutUser } from "../UserAuthentication/Auth";
 import { createPost, fetchUserPosts } from "../UserAuthentication/FirestoreDB";
+import Modal from "../Modal";
 import "./UserProfile.css";
 
 function UserProfilePage({ userUID, userData }) {
+  const navigate = useNavigate();
   const [imageUpload, setImageUpload] = useState(null);
   const [userPosts, setUserPosts] = useState([]);
+  const [isShowModal, setIsShowModal] = useState(false);
   const [isFormValid, setIsFormValid] = useState(false);
+  const [serviceCategories, setSelectedCategories] = useState([]);
+  const [availableCategories] = useState([
+    "Hair Styling",
+    "Plumbing",
+    "Fitness",
+    "Cleaning",
+    "Electricity",
+    "Home Decor",
+    "Car Repair",
+    "Pet Sitting",
+    "Chef",
+  ]);
+  const removeCategory = (category) => {
+    setSelectedCategories(serviceCategories.filter((cat) => cat !== category));
```

Comment:
Rename `cat` to `serviceCategory`
```suggestion
    setSelectedCategories(serviceCategories.filter((serviceCategory) => serviceCategory !== category));
```